vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/DropdownSteps.java

```diff
@@ -104,180 +81,92 @@ public Select isDropDownWithNameFound(String dropDownName)
-        }
-    }
-
-    /**
-     * Checks if a drop down with the specified <b>name</b> does not exist in context
-     * @param dropDownName Any attribute value of the <i>&lt;select&gt;</i> tag
-    */
-    @Then("a drop down with the name '$dropDownName' does not exist")
-    public void doesNotDropDownExist(String dropDownName)
-    {
-        Locator locator = createLocator(dropDownName);
-        locator.getSearchParameters().setVisibility(Visibility.ALL);
-        baseValidations.assertIfElementDoesNotExist(String.format(DROP_DOWN_WITH_NAME, dropDownName), locator);
-    }
-
-    /**
-     * Checks that previously set searchContext contains a drop down with the expected <b>name</b>
-     * and expected State
-     * <p>
-     * <b>Drop down</b> is a drop down list with the specified options;
-     * </p>
-     * <p>
-     * Is created by the <i>&lt;select&gt;</i> tag. The <i>&lt;option&gt;</i> tags inside the <i>
-     * &lt;select&gt;</i> define the available options in the list.</p>
-     * <b>Example:</b>
-     * <pre>
-     *  &lt;select&gt;
-     *   &lt;option&gt;visible text 1&lt;/option&gt;
-     *   &lt;option&gt;visible text 2&lt;/option&gt;
-     *   &lt;option&gt;visible text 3&lt;/option&gt;
-     *  &lt;/select&gt;
-     * </pre>
-     * @param state A state value of the link
-     * (<i>Possible values:</i> <b>ENABLED, DISABLED, SELECTED, NOT_SELECTED, VISIBLE, NOT_VISIBLE, SINGLE_SELECT,
-     * MULTI_SELECT</b>)
-     * @param dropDownName Text value of any attribute of the drop down tag
-     */
-    @Then("a [$state] drop down with the name '$dropDownName' exists")
-    public void isDropDownWithNameFound(DropDownState state, String dropDownName)
-    {
-        Select select = findDropDownList(dropDownName);
-        baseValidations.assertElementState("The found drop down is " + state, state, select);
+        });
     }
 
     /**
-     * Checks whether default value of a drop-down with the specific name
+     * Checks whether default value of a dropdown located by locator
      * is similar to expected
-     * @param dropDownName Text value of any attribute of the drop down tag
-     * @param dropDownText Text that is displayed
+     * @param locator Locator to locate dropdown list
+     * @param option Text that is displayed
      */
-    @Then(value = "a drop down with the name '$dropDownName' and text '$dropDownText' exists", priority = 1)
-    public void ifDropDownWithNameExists(String dropDownName, String dropDownText)
+    @Then(value = "dropdown located `$locator` exists and selected option is `$option`", priority = 1)
+    public void doesDropdownHasFirstSelectedOption(Locator locator, String option)
     {
-        Select dropDown = isDropDownWithNameFound(dropDownName);
-        if (dropDown != null && descriptiveSoftAssert.assertTrue("Selected options are present in drop down",
-                !dropDown.getAllSelectedOptions().isEmpty()))
+        Select dropdown = findDropdown(locator).orElse(null);
```

Comment:
findDropdown(locator).ifPresent(dd -> {  if (softAssert.asse...) } ) to reduce assignment and null check